src/components/ha-picker-combo-box.ts

```diff
@@ -232,7 +235,7 @@ export class HaPickerComboBox extends LitElement {
     const searchString = ev.detail.value.trim() as string;
 
     const index = this._fuseIndex(this._items);
-    const fuse = new HaFuse(this._items, {}, index);
+    const fuse = new HaFuse(this._items, { shouldSort: false }, index);
```

Comment:
I think this was the intention?
```suggestion
    const fuse = new HaFuse(this._items, { shouldSort: !this.noResultSorting }, index);
```